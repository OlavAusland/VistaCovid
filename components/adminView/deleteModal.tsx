import { Alert, Button, Modal, SafeAreaView, ScrollView, TextInput, View, Text, Pressable } from "react-native";
import React, { useEffect, useState } from "react";
import { adminStyle } from "../../styles/AdminStyles";
import {deleteRoom} from "../../api/firebaseAPI";

type DeleteRoomModal={
    modalVisible: boolean;
    handleRequestClose: Function;
    deleteRoomAndClose: Function;
}

export const DeleteRoomModal = (props: DeleteRoomModal) => {
    
    return(
        <Modal
        animationType="slide"
        statusBarTranslucent={true}
        transparent={true}
        visible={props.modalVisible}
        onRequestClose={()=>props.handleRequestClose()}>

                <View style={adminStyle.centeredView}>
                    <View style={adminStyle.modalView}>
                        <Text style={adminStyle.modalText}>Are you sure you want to delete this room?</Text>
                        
                        <Pressable
                        style={[adminStyle.button, adminStyle.buttonClose]}
                        onPress={()=> props.deleteRoomAndClose()}
                        >
                            <Text style={adminStyle.textStyle}>Yes, delete room</Text>
                        </Pressable>

                        <Pressable
                        style={[adminStyle.button, adminStyle.buttonClose]}
                        onPress={()=> props.handleRequestClose()}
                        >
                            <Text style={adminStyle.textStyle}>No, close modal</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
    );
}

