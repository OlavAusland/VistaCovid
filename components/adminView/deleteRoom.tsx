import { ScrollView } from "react-native";


export function DeleteRoom()
{
    return(
        <ScrollView></ScrollView>

    );
}