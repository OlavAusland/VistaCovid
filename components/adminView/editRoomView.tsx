import { ScrollView } from "react-native";


export function EditRoom()
{
    return(
        <ScrollView></ScrollView>

    );
}