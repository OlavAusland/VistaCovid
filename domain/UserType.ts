export enum Roles {
    NONE,
    NURSE,
    DOCTOR,
    ADMIN                      
};

export type User = {
    email: string,
    password: string,
    code: string | undefined | null,
    firstName: string,
    lastName: string,
    role: Roles,
    phone: string | undefined,
    address: string | undefined,
    city: string | undefined,
    id: string
}

export type SimpleUser = {
    displayName:string,
    role: Roles
}

export type LoginInfo = {
    email: string,
    password: string,
    displayPassword: boolean
}

export type currentUser = {
    email: string,
    firstName: string, 
    lastName: string,
    role: Roles,
    id: string
}