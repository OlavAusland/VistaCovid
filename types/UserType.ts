export enum Roles {
    ADMIN,                      
    DOCTOR,
    NURSE,
    NONE
};


export type User = {
    email: string,
    password: string,
    code: string | undefined | null,
    firstName: string,
    lastName: string,
    role: Roles,
    phone: string | undefined,
    address: string | undefined,
    city: string | undefined,
}

export type LoginInfo = {
    email: string,
    password: string,
    displayPassword: boolean
}