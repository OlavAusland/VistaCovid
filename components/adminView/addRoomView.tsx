import { useState } from "react";
import React, { ScrollView, TextInput, View, Text, Button, SafeAreaView, Alert } from "react-native";
import { adminStyle } from "../../styles/AdminStyles";
import { Room } from "../../domain/RoomType";
import { addRoom, getRooms } from "../../api/firebaseAPI";
import { NoteData } from "../../domain/RoomType";
import { Roles } from "../../domain/UserType";

export function AddRoom()
{

    const [roomNumber, setRoomNumber] = useState("")

 
    const newroom: Room = {
        id: roomNumber,
        patientId: "",
        roomNumber: roomNumber,
        lastUpdated: Date.now().toString(),
        heartRate: [],
        bloodPressure: [],
        oxygenLevel: [],
        notes: []
     }


     const SuccessfullAlert = () =>
     Alert.alert(
       "Successfully added room with room number: "+{roomNumber},
       "",
       [
         { text: "OK", onPress: () => console.log("OK Pressed", roomNumber) }
       ]
     );

     const UnsuccessfullAlert = () =>
     Alert.alert(
       "Could not add room with room number: ",
       "Room might already exist",
       [
         { text: "OK", onPress: () => console.log("OK Pressed") }
       ]
     );

     const AddRoom = () =>{
         addRoom(newroom).then(() => {SuccessfullAlert()}).catch(() => {UnsuccessfullAlert()})
     }

    return(
        <ScrollView style={adminStyle.addRoomContainer}>
            <SafeAreaView>
                <View style={adminStyle.addRoomHeader}>
                    <Text style={adminStyle.headertext}>Add Room</Text>
                </View>
                <View>
                    <TextInput
                    style={adminStyle.addRoomInput}
                    onChangeText={setRoomNumber}
                    placeholder="room number"/>
                    <View style={adminStyle.addRoomButton}>
                        <Button title='Add Room' onPress={()=>{AddRoom()}}/>
                    </View>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}