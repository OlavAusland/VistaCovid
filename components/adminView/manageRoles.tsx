
// navigation
import { useEffect, useState } from "react";
import { SafeAreaView, ScrollView, Text, TextInput, TouchableOpacity, View } from "react-native";
import Icon from "react-native-vector-icons/Feather";
import { deleteUserById, getUsers } from "../../api/firebaseAPI";
import { ErrorType } from "../../domain/Errortype";
import { User } from "../../domain/UserType";
import { manageRolesStyles } from '../../styles/ManageRolesStyles';
import { Errormodal } from "../ErrorModal";




export function ManageRoles(){

    const [users, setUsers] = useState<User[]>([])
    const [search, setSearch] = useState<string>('');
    const [error, setError] = useState<ErrorType>({errorObject:undefined, errormodalVisible:false});

    const handleRequestClose = () => {
        setError((prev) =>({...prev,errorObject:undefined, errormodalVisible:false}));
    }

    useEffect(() => {
        const getUserData = async () =>{
            await getUsers().then((res) => {
                setUsers(res);
            }).catch((err) => { setError((prev) =>({...prev, errorObject:err, errormodalVisible:true})); });
        };
        getUserData();
    }, []);

    if(error.errormodalVisible){
        return (
            <Errormodal error={error} handleRequestClose={handleRequestClose} />
        )
    }

    return(
        <SafeAreaView style={{flex:1}}>
            <View style={[manageRolesStyles.header, manageRolesStyles.shadow]}>
                <TextInput
                    placeholder="room number"
                    onChangeText={(text) => {setSearch(text)}}
                    style={manageRolesStyles.search}/>
            </View>
            <ScrollView contentContainerStyle={{flex:1, alignItems:'center'}}>
                {users.map((user) => {
                    return(
                        <View  key={'users' + user.id} style={manageRolesStyles.cardContainer}>
                            <View style={manageRolesStyles.card}>
                                <Icon style={{alignSelf:'center'}} name='clipboard' size={80}/>
                                <Text style={{flex:1}}>Patient: {user.id}</Text>
                            </View>
                            <TouchableOpacity onPress={() =>{deleteUserById(user.id)}} style={manageRolesStyles.delete}>
                                <Icon name='trash-2' color={'white'} size={40}/>
                            </TouchableOpacity>
                        </View>
                    );
                })}
            </ScrollView>
        </SafeAreaView>
    );
}
