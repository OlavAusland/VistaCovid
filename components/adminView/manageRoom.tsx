import { Alert, Button, Modal, SafeAreaView, ScrollView, TextInput, View, Text, Pressable } from "react-native";
import { useState } from "react";
import { deleteRoom } from "../../api/firebaseAPI";
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { StackParameters } from "../../domain/NavigationTypes";
import { adminStyle } from "../../styles/AdminStyles";
import  handleDelete from "./deleteModal"

export function ManageRoom()
{
    const navigation = useNavigation<NativeStackNavigationProp<StackParameters>>();

    const [roomNumber, setRoomNumber] = useState('')
    const [delroomNumber, setDelRoomNumber] = useState('')



    return(
        <ScrollView>
            <SafeAreaView>
                <View style={adminStyle.editRoom}>
                    <TextInput
                    placeholder="room number"
                    onChangeText={setRoomNumber}/>
                    <View style={adminStyle.manageRoomButtons}>
                        <Button title="Edit room" onPress={()=>navigation.navigate("EditRoom")}/>
                    </View>
                </View>

                <View style={adminStyle.deleteRoom}>
                    <TextInput
                    placeholder="room number"
                    onChangeText={setDelRoomNumber}/>
                    <Button title="DeleteRoom" onPress={()=>handleDelete}/>
                </View>
            </SafeAreaView>
        </ScrollView>
    );
}