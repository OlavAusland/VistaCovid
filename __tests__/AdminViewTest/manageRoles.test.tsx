
import { render } from '@testing-library/react-native';
import { ManageRoles } from "../../components/adminView/manageRoles";
import { Roles } from '../../domain/UserType';

const mockUser = [{
    email: 'test@uia.no',
    password: 'testtest',
    code: undefined,
    firstName: 'test',
    lastName: 'testesen',
    role: Roles.NONE,
    phone: '98767564',
    address: 'testgata 4',
    city: 'oslo',
    id: '23456787656'
},
{
    email: 'test2@uia.no',
    password: 'testtest2',
    code: undefined,
    firstName: 'test2',
    lastName: 'testesen2',
    role: Roles.NONE,
    phone: '98767598',
    address: 'testgata 8',
    city: 'oslo',
    id: '23456787856'
}]
test("renders correctly", () => {
    const tree = render(<ManageRoles/>).toJSON();
    expect(tree).toMatchSnapshot();
});

/* jest.mock("../../api/firebaseAPI");
// Tests with no given range, returns all data for a given room
test("Should disaplay users", async () => {
    
    // @ts-ignore
    getUsers.mockResolvedValueOnce(mockUser);

    const tree = render(<ManageRoles />).toJSON();
    expect(tree).toContain('');

}); */
