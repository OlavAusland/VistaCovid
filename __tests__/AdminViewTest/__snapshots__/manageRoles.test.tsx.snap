// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "paddingBottom": 10,
          "paddingTop": 40,
        },
        Object {
          "elevation": 6,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.27,
          "shadowRadius": 4.65,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="room number"
      style={
        Object {
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
        }
      }
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
